transactor/src/main/scala/io/mediachain/datastore/BinaryDatastore.scala

```diff
@@ -0,0 +1,39 @@
+package io.mediachain.datastore
+
+import cats.data.Xor
+import io.mediachain.multihash.MultiHash
+import io.mediachain.types.Datastore._
+import io.mediachain.types.CborSerialization
+
+abstract class BinaryDatastore extends Datastore {
+  override def get(ref: Reference): Option[DataObject] = {
+    ref match {
+      case MultihashReference(multihash) => decode(get(multihash))
+      case _ => None
+    }
+  }
+
+  def get(key: MultiHash): Option[Array[Byte]]
+  
+  def decode(opt: Option[Array[Byte]]): Option[DataObject] = {
+    opt.map { bytes =>
+      CborSerialization.dataObjectFromCborBytes(bytes) match {
+        case Xor.Right(obj) => obj
+        case Xor.Left(err) => {
+          // this is probably indicative of a bug, so throw
+          throw new DatastoreException("Object deserialization error: " + err.message)
+        }
+      }
+    }
+  }
+  
+  override def put(obj: DataObject): Reference = {
```

Comment:
the underlying binary put doesn't return anything new -- the reference is constructed by the BinaryDatastore and can be returned here.
I have it returning no useful results so that the Multihash reference construction happens in one place, in the BinaryDatastore implementation.